### DISCUSSION TITLE
Loading the same url with #... doesn't update webpage

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3703] [General]
I have the following scenario:
1. Load url https://www.calculator.net/random-number-generator.html?slower=1&supper=100&ctype=1&s=5714&submit1=Generate
2. Load url https://www.calculator.net/random-number-generator.html?slower=1&supper=100&ctype=1&s=5714&submit1=Generate**#blablabla**

And the second url is not loaded (webpage is not updated). If it would be the same URL (without #) - it would load ok. If I would load URL with # as the first one - it would work completely OK. How I can fix that or any workaround (like loading between them smth like localhost)?

**Comment by amaitland** [comment]
> And the second url is not loaded (webpage is not updated). If it would be the same URL (without #) - it would load ok

Sounds like the behaviour I would expect. Adding/modifying the url fragment doesn't reload a web page. If you try loading your Urls in `Chrome` you should see the same behaviour.

Let's say you perform the following:
- Open `Chrome`-
- Load https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting via address bar
- Load https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#unmanaged-crashes-when-the-process-dies  via address bar

Upon completing the second load using the address bar you should have navigated within the document. Obviously if your page is a `SPA` it might look like it's loading new, it will be updating the `html/css` dynamically using javascript.

> How I can fix that or any workaround (like loading between them smth like localhost)?

It doesn't sound like there is a bug to me, so nothing to fix exactly. You can load a [Data Uri](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) to force a navigation. e.g. `data:,Hello%2C%20World%21`
